triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -532,6 +555,57 @@ object Server {
     def logTriggerStarted(m: TriggerStarted): Unit =
       server.logTriggerStatus(m.triggerInstance, "running")
 
+    def refreshAccessToken(triggerInstance: UUID): Future[Unit] =
+      for {
+        // Lookup running trigger
+        runningTrigger <- dao.getRunningTrigger(triggerInstance).flatMap {
+          case Some(trigger) => Future.successful(trigger)
+          case None =>
+            Future.failed(new RuntimeException(s"Unknown trigger $triggerInstance"))
+        }
+        // Request a token refresh
+        authUri <- authConfig match {
+          case NoAuth =>
+            throw new RuntimeException("Cannot refresh token without authorization service")
+          case AuthMiddleware(uri) => Future.successful(uri)
+        }
+        refreshToken <- runningTrigger.triggerRefreshToken match {
+          case Some(token) => Future.successful(token)
+          case None =>
+            Future.failed(new RuntimeException(s"No refresh token for $triggerInstance"))
```

Comment:
We should start throwing proper exception types instead of using `RuntimeException` everywhere. Doesn’t have to be in this PR.